### PR TITLE
Build Sphinx Documentation Using GitHub Actions

### PR DESCRIPTION
**Pull request by smokestacklightnin**
### :pencil: Description

**Type:** :memo: `documentation`

This adds the GitHub workflow to build the Sphinx documentation. The documentation currently builds successfully, but is not published yet. 

### :vertical_traffic_light: Testing

How did you test these changes?

[Here is a link to the rendered website](https://smokestacklightnin.github.io/stardis/index.html) It is based on whatever I push most recently (on any branch). I would like to have separate instances for separate branches but don't know how.

- [ ] Testing pipeline
- [X] Building pipeline for documentation

### :ballot_box_with_check: Checklist

- [ ] I requested two reviewers for this pull request
- [ ] I updated the documentation according to my changes

### CODE REVIEW

**Comment by smokestacklightnin** [comment]
Sorry this has so many commits. I had to push a new commit each time I wanted to run the workflow. It might be useful to come up with a better method in the future.

**Comment by jaladh-singhal** [comment]
> Sorry this has so many commits. I had to push a new commit each time I wanted to run the workflow. It might be useful to come up with a better method in the future.

We squash commits, before merging

Can you please put link of the build in PR description so that we can test docs

File: .github/workflows/build-docs.yml
Comment (jvshields) [inline .github/workflows/build-docs.yml:11]: We can probably get rid of this commented out hello world job

File: .github/workflows/build-docs.yml
Comment (jvshields) [inline .github/workflows/build-docs.yml:41]: Maybe this is standard and I'm not aware, but why do we build the environment from a cached lockfile and then update it from tardis releases, rather than just building it from downloaded file in the first place?

**Review by jvshields — COMMENTED** [review]
I left a couple comments, but this generally looks good to me. I agree with Jaladh that you should quickly attach a finished HTML output by the workflow that we can look at, but other than that stand the small changes I think it's good.

**Comment by smokestacklightnin** [comment]
I have squashed 28 commits into 4. Should I squash them all into one?

**Comment by wkerzendorf** [comment]
We will do the squashing - there is no need for you to do it. Your PRs should be containing one idea - so that one squashed commit makes sense.

File: .github/workflows/build-docs.yml
Comment (atharva-2001) [inline .github/workflows/build-docs.yml:9]: why `bash -le`?

File: .github/workflows/build-docs.yml
Comment (atharva-2001) [inline .github/workflows/build-docs.yml:12]: perhaps think about triggers?

File: .github/workflows/build-docs.yml
Comment (atharva-2001) [inline .github/workflows/build-docs.yml:42]: Hardcoded values- can we do something about this

File: .github/workflows/build-docs.yml
Comment (atharva-2001) [inline .github/workflows/build-docs.yml:59]: workflow dispatch isn't defined as a trigger

File: .github/workflows/build-docs.yml
Comment (atharva-2001) [inline .github/workflows/build-docs.yml:82]: where is dest-dir?

File: .github/workflows/build-docs.yml
Comment (atharva-2001) [inline .github/workflows/build-docs.yml:2]: workflow dispatch and pull request events

**Review by jvshields — APPROVED** [review]
It's hard for me to tell that everything will work until deployed, but at this state I'm happy to merge and see if things run on the main branch.

**Review by wkerzendorf — APPROVED** [review]

**Pull request merged**
